eskip/eskip.go

```diff
@@ -44,13 +53,28 @@ type parsedRoute struct {
 	backend  string
 }
 
+// A CustomPredicate object represents a parsed, in-memory, route matching predicate
+// that is defined by extensions.
+type CustomPredicate struct {
```

Comment:
it's new code. It can be renamed to Predicate, not an issue. What would be too big, is unifying the already existing predicates into the same general Predicate type. A good idea, but a separate pr.